src/jit/inline.cpp

```diff
@@ -1247,31 +1254,34 @@ InlineContext* InlineStrategy::NewFailure(GenTreeStmt* stmt, InlineResult* inlin
     InlineContext* parentContext = stmt->gtInlineContext;
     assert(parentContext != nullptr);
     InlineContext* failedContext = new (m_Compiler, CMK_Inlining) InlineContext(this);
+    GenTreeCall*   originalCall  = inlineResult->GetCall();
 
     // Pushing the new context on the front of the parent child list
     // will put siblings in reverse lexical order which we undo in the
     // dumper.
-    failedContext->m_Parent      = parentContext;
-    failedContext->m_Sibling     = parentContext->m_Child;
-    parentContext->m_Child       = failedContext;
-    failedContext->m_Child       = nullptr;
-    failedContext->m_Offset      = stmt->gtStmtILoffsx;
-    failedContext->m_Observation = inlineResult->GetObservation();
-    failedContext->m_Callee      = inlineResult->GetCallee();
-    failedContext->m_Success     = false;
+    failedContext->m_Parent        = parentContext;
+    failedContext->m_Sibling       = parentContext->m_Child;
+    parentContext->m_Child         = failedContext;
+    failedContext->m_Child         = nullptr;
+    failedContext->m_Offset        = stmt->gtStmtILoffsx;
```

Comment:
While you are here, could you please fix headers/signature for 
`InlineContext* InlineStrategy::NewSuccess(InlineInfo* inlineInfo)` 
and
 `InlineContext* InlineStrategy::NewFailure(GenTreeStmt* stmt, InlineResult* inlineResult)`, 
they probably should be the same (and the headers are), but `NewSuccess` doesn't have `stmt       - statement containing call being inlined` as argument. 

Then maybe it will be obvious how to extract this two similar initialization lists into one method.

Reply:
I'll update the comment, but are you suggesting common out some of the work these two methods do? 

It may be clunky as they don't have access to the same data.

Reply:
I think you can delete `GenTreeStmt* stmt` arg from `NewFailure` and do what `NewSuccess` does:
`GenTreeStmt*   stmt          = inlineInfo->iciStmt;`
`calleeContext->m_Offset        = stmt->AsStmt()->gtStmtILoffsx;` <- looks like we do not need this cast here, it is already a statement.

Reply:
We currently don't have access to the `InlineInfo` for the failure cases.

I'll remove the cast too.